docs/source/starthere/fundamentals.rst

```diff
@@ -0,0 +1,240 @@
+NeMo fundamentals
+=================
+
+On this page we will go a little bit deeper on how NeMo works, to give you a good foundation for using NeMo for your :ref:`desired usecase <where_next>`.
+
+NeMo Models
+-----------
+
+A NeMo "model" includes all of the below components wrapped into a singular, cohesive unit:
+
+* neural network architecture,
+
+* dataset & data loaders,
+
+* data preprocessing & postprocessing,
+
+* optimizer & schedulers,
+
+* any other supporting infrastructure: tokenizers, language model configuration, data augmentation etc.
+
+NeMo models are based on PyTorch. Many of their components are subclasses of ``torch.nn.Module``. NeMo models use PyTorch Lightning (PTL) for training, thus reducing the amount of boilerplate code needed.
+
+NeMo models are also designed to be easily configurable; often this is done with YAML files. Below we show simplified examples of a NeMo model defined in pseudocode, and a config defined in YAML. We highlight the lines where the Python config parameter is read from the YAML file.
+
+.. list-table:: Simplified examples of a model and config.
+    :widths: 1 1
+    :header-rows: 0
+
+    * - .. code-block:: python
+	  :caption: NeMo model definition (Python pseudocode)
+	  :linenos:
+	  :emphasize-lines: 4, 7, 10, 13, 16, 20
+
+	  class ExampleEncDecModel:
+	      # cfg is passed so it only contains "model" section
+	      def __init__(self, cfg, trainer):
+	          self.tokenizer = init_from_cfg(cfg.tokenizer)
+
+
+	          self.encoder = init_from_cfg(cfg.encoder)
+
+
+	          self.decoder = init_from_cfg(cfg.decoder)
+
+
+	          self.loss = init_from_cfg(cfg.loss)
+
+
+		  # optimizer configured via parent class
+
+
+	      def setup_training_data(self, cfg):
+	          self.train_dl = init_dl_from_cfg(cfg.train_ds)
+
+	      def forward(self, batch):
+	          # forward pass defined,
+		  # as is standard for PyTorch models
+	          ...
+
+	      def training_step(self, batch):
+	          log_probs = self.forward(batch)
+	          loss = self.loss(log_probs, labels)
+	          return loss
+
+
+      - .. code-block:: yaml
+	  :caption: Experiment config (YAML)
+	  :linenos:
+	  :emphasize-lines: 4, 7, 10, 13, 16, 20
+
+	  #
+	  # desired configuration of the NeMo model
+	  model:
+	      tokenizer:
+	       ...
+
+	      encoder:
+	       ...
+
+	      decoder:
+	       ...
+
+	      loss:
+	       ...
+
+	      optim:
+	       ...
+
+
+	      train_ds:
+	       ...
+
+	  # desired configuration of the
+	  # PyTorch Lightning trainer object
+	  trainer:
+	      ...
+
+
+Configuring and training NeMo models
+------------------------------------
+
+During initialization of the model, a lot of key parameters are read from the config (``cfg``), which gets passed in to the model construtor (left panel above, line 2).
```

Comment:
`a lot of key parameters`; and what about the rest? hard coded in the code?

Reply:
I mean something like "all the parameters that need to be configurable are configurable". Will rephrase

Reply:
Rephrased by getting rid of "a lot of"